Fatal error: THC/THC.h: no such file or directory

## Issue description

I am trying to build from source with NO_DISTRIBUTED=1, and this error appears and the build is terminated. [Here](https://gist.github.com/vishwakftw/cf8d17f1e8dd50866488df9bdb4ca175) is the complete compilation log.

This occurred with a fresh clone of my fork https://github.com/vishwakftw/pytorch

## System Info
- PyTorch or Caffe2: PyTorch
- How you installed PyTorch (conda, pip, source): source
- Build command you used (if compiling from source): python setup.py build develop
- OS: Ubuntu 16.04.2
- Python version: 3.6.5 (Miniconda)
- CUDA/cuDNN version: 8.0.44
- GPU models and configuration: GeForce 940M
- GCC version (if compiling from source): 5.4.0
- CMake version: 3.11.3

Attempting to install without NO_DISTRIBUTED=1 leads to this error 

```
CMake Error at /media/vishwak/Official/Projects/pytorch/cmake/Modules_CUDA_fix/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find CUDA (missing: CUDA_NVCC_EXECUTABLE)
```

Full log [here](https://gist.github.com/vishwakftw/cae09ec524caa523fdd5574d49a31914) .

--- a/tools/build_pytorch_libs.sh
+++ b/tools/build_pytorch_libs.sh
@@ -226,7 +226,9 @@ function build_nccl() {
   popd
 
   # FIXME: sccache doesn't work when building NCCL
-  export CUDA_NVCC_EXECUTABLE=$CUDA_NVCC_EXECUTABLE_SAVED
+  if [ -n "$CUDA_NVCC_EXECUTABLE_SAVED" ]; then
+    export CUDA_NVCC_EXECUTABLE=$CUDA_NVCC_EXECUTABLE_SAVED
+  fi
   export PATH=/opt/cache/bin:$PATH
 }
 

